frb_codegen/src/generator/rust/ty_enum.rs

```diff
@@ -195,9 +195,15 @@ impl TypeRustGeneratorTrait for TypeEnumRefGenerator<'_> {
                         let fields = Some(tag)
                             .into_iter()
                             .chain(st.fields.iter().map(|field| {
-                                let gen =
-                                    TypeRustGenerator::new(field.ty.clone(), self.context.ir_file);
-                                gen.convert_to_dart(field.name.rust_style().to_owned())
+                                if field.name.raw == "backtrace" {
```

Comment:
That does not look perfectly neat... We are treating struct `Backtrace` as sth very special here... So maybe we can PR to allo-isolate to support backtrace.

Reply:
The only way to know it's really a backtrace is to have a field name backtrace. I also don't see how allo-isolate would solve this

Reply:
Well, I mean, for a backtrace, we do:

```
"format!(\"{:?}\", backtrace).into_dart()".to_string()
```

For all other types, we do:

```
gen.convert_to_dart(field.name.rust_style().to_owned())
```

Thus backtrace is so special here that we write a brand new codegen rule for it...

Reply:
I tried adding a type delegate for backtrace, but this part on src/generator/rust/ty.rs

```rust
    fn convert_to_dart(&self, obj: String) -> String {
        format!("{}.into_dart()", obj)
    }
```

generates

```rust
impl support::IntoDart for CustomError {
    fn into_dart(self) -> support::DartCObject {
        match self {
            Self::Error0 { e, backtrace } => {
                vec![0.into_dart(), e.into_dart(), backtrace.into_dart()]
            }
            Self::Error1 { e, backtrace } => {
                vec![1.into_dart(), e.into_dart(), backtrace.into_dart()]
            }
        }
        .into_dart()
    }
}
```

but of course `backtrace.into_dart()` does not work, it should be `format!("{:?}", backtrace).into_dart()`, but it's hard to do this modification, it will require reengineering a lot of stuff

Reply:
> but of course backtrace.into_dart() does not work, it should be format!("{:?}", backtrace).into_dart()

Can we `impl IntoDart for Backtrace`?

Reply:
I think this is too arbitrary to request for the backtrace developers, and once the official backtrace support lands, it will not have a into dart implementation

Reply:
Well I mean PR to allo-isolate, not PR to backtrace package

Reply:
https://github.com/sunshine-protocol/allo-isolate/pull/25